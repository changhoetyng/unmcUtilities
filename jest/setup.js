const rn = require('react-native')

jest.mock('Linking', () => {
    return {
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      openURL: jest.fn(),
      canOpenURL: jest.fn(),
      getInitialURL: jest.fn(),
    }
})

import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock';

jest.mock('@react-native-async-storage/async-storage', () => mockAsyncStorage);

module.exports = rn