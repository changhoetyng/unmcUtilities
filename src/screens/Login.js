import React from 'react';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {
  Text,
  View,
  Image,
  Keyboard,
  TextInput,
  StyleSheet,
  TouchableOpacity,
  KeyboardAvoidingView,
  TouchableWithoutFeedback,
} from 'react-native';
import {theme} from '../styles/ThemeColour';
import FullPageLoader from "../hooks/FullPageLoader"

function Login({navigation}) {
  const pressHandler = () => {
    navigation.navigate('Home');
  };

  return (
    <TouchableWithoutFeedback
      onPress={() => {
        Keyboard.dismiss();
      }}>
      <KeyboardAvoidingView
        style={{flex: 1, alignItems: 'center', backgroundColor: '#ddd'}}
        behavior="position">
          {/* <FullPageLoader /> */}
        <View style={{alignItems: 'center'}}>
          {/* <Image
            source={require('../assets/UNMC-removebg-preview.png')}
            style={styles.imageSize}
          /> */}
        </View>
        <View style={styles.container}>
          <TextInput style={styles.input} placeholder="Email" />

          <TextInput style={styles.input} placeholder="Password" />

          <TouchableOpacity onPress={pressHandler}>
            <View style={styles.loginButton}>
              <Icon name="login" size={18} color="white" />
            </View>
          </TouchableOpacity>

          <TouchableOpacity>
            <Text style={styles.textStyle}> Forget password? </Text>
          </TouchableOpacity>
        </View>
      </KeyboardAvoidingView>
    </TouchableWithoutFeedback>
  );
}

export default Login;

const styles = StyleSheet.create({
  input: {
    width: 200,
    margin: 5,
    padding: 8,
    borderWidth: 1,
    borderRadius: 5,
    borderColor: 'black',
    backgroundColor: 'white'
  },

  container: {
    width: 300,
    padding: 10,
    marginTop: 225,
    alignItems: 'center',
    borderWidth: 1,
    borderRadius: 20,
    backgroundColor: 'white'
  },

  textStyle: {
    color: theme.primary,
    fontSize: 12,
    marginTop: 10
  },

  imageSize: {
    width: 200,
    height: 200,
    resizeMode: 'contain'
  },

  loginButton: {
    width: 200,
    padding: 10,
    alignItems: 'center',
    borderWidth: 1,
    borderStyle: 'solid',
    borderRadius: 5,
    backgroundColor: theme.primary
  },
});
