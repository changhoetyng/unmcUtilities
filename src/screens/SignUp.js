import React, {Component} from 'react';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {
  Text,
  View,
  Image,
  Keyboard,
  TextInput,
  StyleSheet,
  TouchableOpacity,
  Dimensions,
  KeyboardAvoidingView,
  Button,
  TouchableWithoutFeedback,
} from 'react-native';
import {theme} from '../styles/ThemeColour';
import FullPageLoader from '../hooks/FullPageLoader';
import Modal from 'react-native-modal';
import {api} from '../api/api';

class SignUp extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: false,
      data: [],
      studentId: '',
      email: '',
      password: '',
      confirmPassword: '',
      modalSuccess: false
    };
  }

  async signUp() {
    this.setState({loading: true})
    if (!/^[a-zA-Z0-9]+@[a-zA-Z0-9]/.test(this.state.email)) {
        this.state.error = "Invalid email";
    } else if(this.state.password != this.state.confirmPassword) {
        this.state.error = "Password doesn't match confirm password";
    } else {
    await api
      .post('/student/signup', {
        studentId: this.state.studentId,
        email: this.state.email,
        password: this.state.password
      })
      .then(() => {
        this.setState({modalSuccess: true})
      })
      .catch((err) => {
        if (!err.response) {
          this.setState({error: "Error: Network Error"});
        } else {
            this.setState({error: err.response.data.error});
        }
      });
    }
    
      this.setState({loading: false})
  }

  render() {
    const screenWidth = Dimensions.get('window').width;
    return (
      <TouchableWithoutFeedback
        onPress={() => {
          // await dispatch(addToken('TESTING MOM'))
          Keyboard.dismiss();
        }}>
        <KeyboardAvoidingView
          style={{flex: 1, alignItems: 'center', backgroundColor: '#ddd'}}
          behavior="height">
          {this.state.loading && <FullPageLoader />}
          <View style={{flex: 1, justifyContent: 'center'}}>
            <View style={styles.container}>
              <TextInput
                style={styles.input}
                onChangeText={(text) => this.setState({studentId: text})}
                placeholder="Student ID"
              />
              <TextInput
                style={styles.input}
                onChangeText={(text) => this.setState({email: text})}
                placeholder="E-mail"
              />
              <TextInput
                style={styles.input}
                secureTextEntry={true}
                onChangeText={(text) => this.setState({password: text})}
                placeholder="Password"
              />

              <TextInput
                style={styles.input}
                secureTextEntry={true}
                onChangeText={(text) =>this.setState({confirmPassword: text})}
                placeholder="Confirm Password"
              />

              <TouchableOpacity onPress={() => this.signUp()}>
                <View style={styles.loginButton}>
                  <Text style={{color: 'white'}}>Sign Up</Text>
                </View>
              </TouchableOpacity>
              <Text style={{color: "red"}}>{this.state.error}</Text>
              <TouchableOpacity testID = {"backButton"} onPress={() => this.props.navigation.goBack()}>
                <Text style={styles.signUp}> Go Back </Text>
              </TouchableOpacity>
            </View>
          </View>
          <Modal isVisible={this.state.modalSuccess}>
          <View
            style={{
              width: screenWidth * 0.7,
              height: screenWidth * 0.7,
              backgroundColor: 'white',
              alignSelf: 'center',
              flexWrap: 'wrap',
              padding: 10,
              alignContent: 'center',
            }}>
            <View style={{width: '100%', height: '100%'}}>
              <Text
                style={{fontWeight: 'bold', textAlign: 'center', fontSize: 20}}>
                 Successfully signed up.
              </Text>
              <View style={{width: '100%', position: 'absolute', bottom: 0}}>
                <Button
                testID = {"backButtonModal"}
                  title="Go Back"
                  onPress={() => {
                      this.props.navigation.goBack();
                  }}
                />
              </View>
            </View>
          </View>
        </Modal>
        </KeyboardAvoidingView>
      </TouchableWithoutFeedback>
    );
  }
}

export default SignUp;

const styles = StyleSheet.create({
  input: {
    width: 200,
    margin: 5,
    padding: 8,
    borderWidth: 1,
    borderRadius: 5,
    borderColor: 'black',
    backgroundColor: 'white',
  },

  container: {
    width: 300,
    padding: 10,
    alignItems: 'center',
    borderWidth: 1,
    borderRadius: 20,
    backgroundColor: 'white',
  },

  signUp: {
    color: theme.primary,
    fontSize: 13,
    marginTop: 10,
  },

  imageSize: {
    width: 250,
    height: 250,
    resizeMode: 'contain',
  },

  loginButton: {
    width: 200,
    padding: 10,
    alignItems: 'center',
    borderWidth: 1,
    borderStyle: 'solid',
    borderRadius: 5,
    backgroundColor: theme.primary,
  },
});
