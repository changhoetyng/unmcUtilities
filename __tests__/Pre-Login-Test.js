import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import Login from '../src/screens/Login';
import SignUp from '../src/screens/SignUp';
import renderer from 'react-test-renderer';
const middlewares = [];
const mockStore = configureStore(middlewares);

jest.mock('@react-native-community/async-storage', () =>
  require('@react-native-community/async-storage/jest/async-storage-mock'),
);

const navigation = {
  goBack: jest.fn(),
  navigate: jest.fn(),
};
describe('Login Screens', () => {
  jest.useFakeTimers()

  const store = mockStore({});
  jest.mock('assets/UNMC-removebg-preview.png');
  const login = renderer.create(
    <Provider store={store}>
      <Login navigation={navigation}/>
    </Provider>,
  );

  beforeEach(() => {
    jest.resetModules();
    jest.resetAllMocks();
    jest.useFakeTimers();
  });

  it('Login Snapshot', () => {
    expect(login.toJSON()).toMatchSnapshot();
  });

  it('Call Sign Up if Sign Up button is pressed', async () => {
    const button = login.root.findByProps({testID: 'signUpButton'}).props;
    await button.onPress();
    expect(navigation.navigate).toBeCalledWith('SignUp');
  });
});

describe('Sign Up Screen', () => {
  jest.useFakeTimers()

  const store = mockStore({});
  const signUp = renderer.create(
    <Provider store={store}>
      <SignUp navigation={navigation}/>
    </Provider>,
  );

  beforeEach(() => {
    jest.resetModules();
    jest.resetAllMocks();
    jest.useFakeTimers();
  });

  it('SignUp Snapshot', () => {
    expect(signUp.toJSON()).toMatchSnapshot();
  });

  it('Go Back if Back button is pressed', async () => {
    const button = signUp.root.findByProps({testID: 'backButton'}).props;
    await button.onPress();
    expect(navigation.goBack).toHaveBeenCalled();
  });

  it('Go Back if Back button on modal is pressed', async () => {
    const button = signUp.root.findByProps({testID: 'backButtonModal'}).props;
    await button.onPress();
    expect(navigation.goBack).toHaveBeenCalled();
  });
});