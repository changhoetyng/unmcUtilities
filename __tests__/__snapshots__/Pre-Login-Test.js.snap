// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screens Login Snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ddd",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Image
      source={Object {}}
      style={
        Object {
          "height": 250,
          "resizeMode": "contain",
          "width": 250,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
        "marginBottom": 30,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 20,
          "borderWidth": 1,
          "padding": 10,
          "width": 300,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Student ID"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "margin": 5,
            "padding": 8,
            "width": 200,
          }
        }
        underlineColorAndroid="transparent"
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Password"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "margin": 5,
            "padding": 8,
            "width": 200,
          }
        }
        underlineColorAndroid="transparent"
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#70A9FF",
              "borderRadius": 5,
              "borderStyle": "solid",
              "borderWidth": 1,
              "padding": 10,
              "width": 200,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 18,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            î©·
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="signUpButton"
      >
        <Text
          style={
            Object {
              "color": "#70A9FF",
              "fontSize": 13,
              "marginTop": 10,
            }
          }
        >
           Sign Up 
        </Text>
        <Text
          style={
            Object {
              "color": "red",
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Sign Up Screen SignUp Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ddd",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 20,
          "borderWidth": 1,
          "padding": 10,
          "width": 300,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Student ID"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "margin": 5,
            "padding": 8,
            "width": 200,
          }
        }
        underlineColorAndroid="transparent"
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="E-mail"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "margin": 5,
            "padding": 8,
            "width": 200,
          }
        }
        underlineColorAndroid="transparent"
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Password"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "margin": 5,
            "padding": 8,
            "width": 200,
          }
        }
        underlineColorAndroid="transparent"
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Confirm Password"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "margin": 5,
            "padding": 8,
            "width": 200,
          }
        }
        underlineColorAndroid="transparent"
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#70A9FF",
              "borderRadius": 5,
              "borderStyle": "solid",
              "borderWidth": 1,
              "padding": 10,
              "width": 200,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
              }
            }
          >
            Sign Up
          </Text>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "red",
          }
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="backButton"
      >
        <Text
          style={
            Object {
              "color": "#70A9FF",
              "fontSize": 13,
              "marginTop": 10,
            }
          }
        >
           Go Back 
        </Text>
      </View>
    </View>
  </View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      forwardedRef={[Function]}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "margin": 37.5,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "backgroundColor": "white",
            "flexWrap": "wrap",
            "height": 525,
            "padding": 10,
            "width": 525,
          }
        }
      >
        <View
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Successfully signed up.
          </Text>
          <View
            style={
              Object {
                "bottom": 0,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={Object {}}
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="backButtonModal"
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "margin": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Go Back
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
